### PR TITLE
blast-plus: update to 2.13.0 and add cpio as build dependency

### LINKED ISSUE #33148: Installation issue: blast-plus (missing cpio)
### Steps to reproduce the issue

```console
cat > spack.yaml << EOF
spack:
  view: true
  specs:
  - blast-plus@2.12.0
  container:
    extra_instructions:
      final: RUN (echo /opt/view/lib && echo /opt/view/lib64) > /etc/ld.so.conf.d/spack-view.conf
        && ldconfig
    images:
      os: ubuntu:22.04
      spack:
        ref: develop
EOF
spack containerize > Dockerfile
docker build -t blast:2.12.0 .
```

### Error message

```
 59307    done
     59308    /bin/bash: line 3: cpio: command not found
     59309    /bin/bash: line 3: cpio: command not found
  >> 59310    make[1]: *** [Makefile:47: install-toolkit] Error 127
     59311    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-bla
              st-plus-2.12.0-rlog3ww57onydmbgmpbqj25nvcmayses/spack-src/c++'
  >> 59312    make: *** [Makefile:63: install] Error 2
```

### Information on your system

c7g-4x Instance

### Additional information

_No response_

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

### PR DESCRIPTION
**Pull request by ChristianKniep**
To fix #33148 I changed the package and add a build dependency. While at it I updated the version to the latest. 
@weijianwen (Maintainer)

### CODE REVIEW

File: var/spack/repos/builtin/packages/blast-plus/package.py [hunk 1]
```diff
@@ -45,6 +46,8 @@ def patch(self):
     # No...
     # depends_on :mysql => :optional
 
+    depends_on("cpio", type="build")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/blast-plus/package.py:49]: Does this dependency apply for the new version (on) or all versions?

**Review by tldahlgren — APPROVED** [review]
Confirmed sha256.  Have a quick question that could affect the new dependency.

**Comment by tldahlgren** [comment]
Will defer merge decision to give @weijianwen a chance to comment.

**Review by bernhardkaindl — APPROVED** [review]
As it fixes #33148, the dependency on `cpio` can be assumed for for all versions:

Usually: `cpio` is a standard package, so it is not uncommon for this bug to be uncovered only when run on a system without `cpio`, which is rare.

**Pull request merged**